Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -2282,6 +2282,14 @@ LayoutUnit RenderBox::perpendicularContainingBlockLogicalHeight() const
     return cb->adjustContentBoxLogicalHeightForBoxSizing(LayoutUnit(logicalHeightLength.value()));
 }
 
+bool RenderBox::isAdjacentToBlockContainerBeforeEdge() const
+{
+    if (needsLayout())
+        return false;
```

Comment:
this should assert. this function should not be called with stale geometry.

Reply:
Changed and moved to `RenderBox::hasTrimmedMargin`